### ISSUE TITLE
Change time inconsistency in stat command output with from cache or from azure

### ISSUE THREAD

**Issue by akashvrts** [Azure/azure-storage-fuse#291]
### Which version of the blobfuse was used?
1.0.2

### Which OS (please include version) are you using?
RHEL 7.5

### What problem was encountered?
The use case is,
My application compares the change time and modification time of files to identify the changes in file to perform certain optimization.
But the information regarding change time and access time changes with cache expiry.

So the issue is
I have created a file on blobfuse mount.
If i run stat command, it returns valid values for change time, modification time and access time.
However, if i run the same command after the cache expiry time, the modification time is returned with a valid value, but access time and change time are set to EPOCH.

 ```
[root@akash~]# stat /testakash/test.txt
  File: ‘/testakash/test.txt’
  Size: 4               Blocks: 0          IO Block: 4096   regular file
Device: 28h/40d Inode: 8           Links: 1
Access: (0770/-rwxrwx---)  Uid: (    0/    root)   Gid: (    0/    root)
Access: 1970-01-01 05:30:00.000000000 +0530
Modify: 2019-07-18 16:48:03.000000000 +0530
Change: 1970-01-01 05:30:00.000000000 +0530
 Birth: -
```

On similar lines, another observation is
I have mounted my bucket with cache time out equal to 0.
The stat command returns valid values of change time, modify time and access time as long as some process is reading the file. Once the operation is complete, the time gets set to EPOCH.

Is there any workaround to get the change time value consistent. Or can we off the cache at al?

### Have you found a mitigation/solution?

Can we set the modification time to change time as well? Since azure does not return any value related to change time in blob property, we may not actually get it in blob property.

### By default, blobfuse logs errors to syslog. If this is relevant, is there anything in the syslog that might be helpful?
NA

### If relevant, please share your mount command.
blobfuse /mount-dir --tmp-path=/home/blobcahce --config-file=/connection.cfg --log-level=LOG_DEBUG --file-cache-timeout-in-seconds=120 -o remember=-1

**Comment by amnguye** [comment]
We already currently have an issue open for this (#264 ), but thank you for introducing another scenario to reproduce this.

At this time you cannot turn off the cache, and at the same time even if you could it would have the same results as setting the "--file-cache-timeout-in-seconds=0" flag. If anything setting a longer cache will keep the timestamps around longer.

At this time blobfuse doesn't have the ability to change the modification time to the change time on a file.

**Comment by akashvrts** [comment]
When can i expect the solution for this issue?
Do you consider copying "modification time" in "change time" as solution for consistency, since blob property does not have change time field?

**Comment by amnguye** [comment]
I cannot give an exact date on when you expect a solution for this issue. However we will do our best to get this into the next release.

I don't consider making the modification time and the change time the exact same since that is not accurate but it's definitely a bandaid for the solution considering we do not currently save the change time for blobs.

**Comment by vibhansa-msft** [comment]
Timestamp issues have been fixed in version 1.3.1
